### PR TITLE
Scenario framework

### PR DESCRIPTION
**Pull request by kaelyndunnell**
Added CI & testing, built multi-pulse capability as in issue #13, made consistent material values as in issue #19, fixed FENICS error that is soon to be fixed in FESTIM (issue #18).

### CODE REVIEW

File: src/hisp/helpers.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        """Returns the time that has elapsed in scenario up until input row.
+
+        Args:
+            row (int): the row index in the scenario file
+
+        Returns:
+            float: the time that has elapsed in scenario until input row 
+        """
+        time_elapsed = 0
+        for row in range(0,row):
```
Comment (RemDelaporteMathurin) [inline src/hisp/helpers.py:133]: This is a bit confusing. I would have another variable name for the counter here like `prev_row_id`

File: src/hisp/helpers.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    
+    def get_time_till_row(self, row:int) -> float:
+        """Returns the time that has elapsed in scenario up until input row.
+
+        Args:
+            row (int): the row index in the scenario file
+
+        Returns:
+            float: the time that has elapsed in scenario until input row 
+        """
```
Comment (RemDelaporteMathurin) [inline src/hisp/helpers.py:131]: Just by looking at this it is unclear if this is inclusive or exclusive: does it include the "current" row? In other words does it count until the begining of the current `row` or the end of the current row

File: test/test_scenario.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    colors = [color[pulse_type] for pulse_type in pulse_types]
+
+    for i in range(len(times) - 1):
+        plt.plot(times[i : i + 2], np.ones_like(times[i : i + 2]), c=colors[i])
+    # plt.xscale("log")
+    # plt.show()
+
+one_line_scenario_path = os.path.join(current_dir, "one_line_scenario.txt")
```
Comment (RemDelaporteMathurin) [inline test/test_scenario.py:57]: I would put that at the top of the file

File: test/test_scenario.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    assert computed_maximum_time == expected_maximum_time
+
+
+@pytest.mark.parametrize("t, expected_row", [(0, 0), (6000, 1)])
+def test_get_pulse_row(t, expected_row):
+    my_scenario = Scenario(scenario_path)
+
+    pulse_row = my_scenario.get_row(t=t)
+
+    assert pulse_row == expected_row
```
Comment (RemDelaporteMathurin) [inline test/test_scenario.py:29]: You should also test that when given a `t` greater than the maximum time `get_row` returns an error

File: test/test_scenario.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+from src.hisp.helpers import Scenario
+import os
+import pytest
+import numpy as np
+import matplotlib.pyplot as plt
+
+current_dir = os.path.dirname(__file__)
+scenario_path = os.path.join(current_dir, "scenario_test.txt")
```
Comment (RemDelaporteMathurin) [inline test/test_scenario.py:8]: 💡 Instead of making text files manually, we should look into adding the possibility to create a scenario in python (and export it to txt)

Example interface:

```python
from hisp import Scenario

my_scenario = Scenario()
my_scenario.add_pulse(pulse_type="FP", ramp_up=..., flattop=..., nb_pulses=...)
my_scenario.add_pulse(pulse_type="FP", ramp_up=..., flattop=..., nb_pulses=...)

my_scenario.export_to_txt("scenario.txt")
```

This would greatly help testing and reduce manual steps

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    { name = "Kaelyn Dunnell" },
+    { email = "kaelyn@mit.edu" },
+]
+dynamic = ["version"]
+name = "HISP"
+description = "Hydrogen Inventory Simulations of PFCs"
+readme = "README/md"
+requires-python = ">=3.9"
+license = { file = "LICENSE" }
+dependencies = ["h_transport_materials"]
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply pyproject.toml:16]: This is how we'd do it https://chrisholdgraf.com/blog/2022/install-github-from-pyproject/

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -23,6 +23,15 @@ jobs:
...
           environment-file: environment.yml
 
+      - name: install HISP
+        shell: bash -l {0}
+        run: python -m pip install .[test]
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          pytest test/ --cov festim --cov-report xml --cov-report term
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply .github/workflows/ci.yml:33]: Also the coverage is wrong here...

```suggestion
      - name: Run tests
        shell: bash -l {0}
        run: |
          pytest test/ --cov hisp--cov-report xml --cov-report term
```
